examples/pxScene2d/src/pxScene2d.cpp

```diff
@@ -655,6 +655,59 @@ rtError pxObject::moveToBack()
   return RT_OK;
 }
 
+rtError pxObject::moveForward()
+{
+    pxObject* parent = this->parent();
+
+    if(!parent)
+        return RT_OK;
+
+    std::vector<rtRef<pxObject> >::iterator it = parent->mChildren.begin(), it_prev;
+    while( it != parent->mChildren.end() )
+    {
+        if( it->getPtr() == this )
+        {
+            break;
+        }
+        it++;
+    }
+    it_prev = it++;
+    if( it == parent->mChildren.end() )
+        return RT_OK;
+
+    rtRef<pxObject> tmp = *it_prev;
+    *it_prev = *it;
+    *it = tmp;
+
```

Comment:
Let's trigger a repaint and mark the mScene's mDirty flag as true after moving forward.  For example:

parent->repaint();
parent->repaintParents();
mScene->mDirty = true;